### ISSUE TITLE
Missing health bars on Jojo [BLUS31405] on AMD after #4265

### ISSUE THREAD

**Issue by toccata10** [RPCS3/rpcs3#4380]
Archlinux, r9 fury, mesa 17.3.7
After #4265, JoJo's Bizarre Adventure: All-Star Battle [BLUS31405] is missing the top bars. It's a known problem for Nvidia, but it used to work fine on AMD. It disappeared both with Ogl and Vulkan.
Comparison screenshots and logs
![fps 29 81 opengl 0 0 5-6510 jojo s bizarre adventure all-star battle blus31405 _064](https://user-images.githubusercontent.com/7478580/38218905-a0786178-36d3-11e8-8e85-9e950ce5d2e1.jpg)
![fps 28 95 opengl 0 0 5-6631 jojo s bizarre adventure all-star battle blus31405 _063](https://user-images.githubusercontent.com/7478580/38218907-a2852b36-36d3-11e8-9cef-b1bc147d7b44.jpg)
[RPCS3.log-Jojo-OK-6510.gz](https://github.com/RPCS3/rpcs3/files/1869519/RPCS3.log-Jojo-OK-6510.gz)
[RPCS3.log-Jojo-missing-6631.gz](https://github.com/RPCS3/rpcs3/files/1869520/RPCS3.log-Jojo-missing-6631.gz)

**Comment by Asinin3** [comment]
This issue is old, not related to that PR AFAIK

**Comment by toccata10** [comment]
I was disturbed by your comment, so I tested also on Windows 10 (Vulkan), on r9 fury, to narrow down the culprit. The rev6516 is displaying bars, rev6541 is not. So it seems to me it's a new issue, related to the massive #4265.

**Comment by Asinin3** [comment]
People have been reporting that the health bars go missing since before I joined Discord (August 2017), from what I know, the issue is random and only affects some people. Confirmed to affect both Nvidia and OpenGL as far as I know.

**Comment by toccata10** [comment]
On Rev7104, health bars are still missing.
Now that rpcs3 has a RSX capture tool, maybe it could help get this pb solved:
https://www.dropbox.com/s/7bse8jhghjip7nh/capture-JojoAsb.rrc.zip?dl=0

**Comment by Canzah** [comment]
Issue still present in latest build as of today.

**Comment by Fudgecake4567** [comment]
I paused and the health bars show up for a brief second, I was wondering if this could help

**Comment by SutandoTsukai181** [comment]
Commit https://github.com/RPCS3/rpcs3/commit/e8023143 broke the life bars. It's something related to how the game requests depth changes in real time as that's how the life bars work. If you jump you can see your character render in front of the life bar. Trying to directly swap the the old code on latest master made the game crash; needs some proper code.

**Comment by SutandoTsukai181** [comment]
This happened on the build that should be working; life bars are only visible under the light of the corpse parts on the ground.
https://youtu.be/WR6uI0DiZmo

It was working for me just before and I didn't change anything in emu, nor did I update my Nvidia drivers. Also worth noting is that the Farseer build doesn't have life bars on nvidia systems with older drivers (<400)

It probably has something to do with me being on a laptop and hibernating a lot. Either a reboot or a GPU driver update will fix this, but I'm not going to cause this might help fixing the issue for now.

**Comment by kd-11** [comment]
I ram some hardware tests; both the old code (which worked) and the new code (which doesn't work) are wrong. I'll push out a patch to match hardware behaviour soon.

**Comment by kd-11** [comment]
Need updated RSX captures to check if its related.

**Comment by MANGOPELICAN** [comment]
> Need updated RSX captures to check if its related.
@kd-11 

https://mega.nz/#!niBw1S5Q!IVz8y2ees_tnBxfmyuVvVlnEAVEOgS1DpQJwVo7z2mU
Download link with the RSX Captures

**Comment by SutandoTsukai181** [comment]
Master and rsx-volatile Captures (different states with one instance where life bars appear):
https://mega.nz/#!GQMCGIQB!cIBHjYd0rIf6Xo_BzbOZLgEUTmNJxnSIjplW8-U8ylM

Will upload renderdoc captures from spu-perf soon, when the life bars were working

**Comment by SutandoTsukai181** [comment]
I didn't specify when exactly did the life bars suddenly appear on master/rsx-volatile, so here are 2 quick videos showing the process on both spu-perf and rsx-volatile:

[spu-perf](https://www.youtube.com/watch?v=7M1KvVDDXNo&feature=youtu.be)
[rsx-volatile(master is the same)](https://www.youtube.com/watch?v=IVQSOAKBXfA&feature=youtu.be)

It happens at the end of a specific character's super.

**Comment by kd-11** [comment]
Its some condition inversion in the shader, I'm running final tests right now.

**Comment by kd-11** [comment]
Also, even spu-perf was technically broken, not sure why it was working but I'll find out shortly.

**Comment by kd-11** [comment]
The bug had nothing to do with shaders; its properly fixed with [this commit.](https://github.com/kd-11/rpcs3/commit/2a21c0fc2af2083ac61e268b6582ac191c12374c). The changeset is too large to include in misc fixes PR so you'll have to wait. For builds, see https://ci.appveyor.com/project/kd-11/rpcs3/builds/26835227.

**Comment by SutandoTsukai181** [comment]
Confirmed working with OpenGL on default settings
![unknown](https://user-images.githubusercontent.com/52977072/63381329-1e7a6700-c3a1-11e9-8641-0cf11c70d3f4.png)

Will try Vulkan and update
